test/acceptance/worker-test.js

```diff
@@ -0,0 +1,597 @@
+import { vi } from 'vitest';
+import fs from 'node:fs';
+import path from 'node:path';
+
+import { setupProject, teardownProject, runBin } from '../helpers/bin-tester.js';
+import setupEnvVar from '../helpers/setup-env-var.js';
+
+vi.setConfig({ testTimeout: 10_000 });
+
+describe('ember-template-lint worker threads', function () {
+  setupEnvVar('FORCE_COLOR', '0');
+  setupEnvVar('LC_ALL', 'en_US');
+
+  let project;
+
+  beforeEach(async function () {
+    project = await setupProject();
+    await project.chdir();
+  });
+
+  afterEach(function () {
+    teardownProject();
+  });
+
+  describe('worker thread activation', function () {
+    it('should use worker threads when file count exceeds threshold', async function () {
+      // MIN_FILES_TO_USE_WORKERS = 100, so we need to create at least 100 files
+      await project.setConfig({
+        rules: {
+          'no-bare-strings': 'error',
+        },
+      });
+
+      // Create 101 template files with errors to trigger worker usage
+      const templateFiles = {};
+      for (let i = 0; i < 101; i++) {
+        templateFiles[`template-${i}.hbs`] = `<div>Bare string ${i}</div>`;
+      }
+
+      await project.write({
+        app: {
+          templates: templateFiles,
+        },
+      });
+
+      let result = await runBin('.');
+
+      expect(result.exitCode).toEqual(1);
+      expect(result.stdout).toContain('Linting 101 Total Files with TemplateLint');
+      expect(result.stdout).toContain('✖ 101 problems (101 errors, 0 warnings)');
+      expect(result.stderr).toBeFalsy();
+    });
+
+    it('should not use worker threads when file count is below threshold', async function () {
```

Comment:
how do we know a worker was used or not? or how many workers?

Reply:
Finally, I figured out, tests updated